src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -113,9 +116,20 @@ class SolverReport:
     best_cost: int = None
     best_state: np.ndarray = None
     best_timestep: int = None
+    cost_timeseries: np.ndarray = None
     solver_config: SolverConfig = None
     profiler: Profiler = None
 
+    def plot_cost_timeseries(self, filename: str = None) -> None:
+        if self.cost_timeseries is None:
+            return NotImplemented  # what to do?

```

Comment:
We shouldn't leave this question in the final code. Either we go with the current implementation, or we implement it. But this question looks like immature code.